src/pybind/mgr/cephadm/tests/test_spec.py

```diff
@@ -70,10 +68,47 @@ def test_spec_octopus():
   "rgw_realm": "default-rgw-realm",
   "rgw_zone": "eu-central-1",
   "subcluster": "1"
+},
+{
+  "service_type": "osd",
+  "service_id": "osd_spec_default",
+  "placement": {
+    "host_pattern": "*"
+  },
+  "data_devices": {
+    "model": "MC-55-44-XZ"
+  },
+  "db_devices": {
+    "model": "SSD-123-foo"
+  },
+  "wal_devices": {
+    "model": "NVME-QQQQ-987"
+  }
 }
 ]
-"""
-    dds_text = """[
+""")
+)
+def test_spec_octopus(spec_json):
+    # https://tracker.ceph.com/issues/44934
+    # Those are real user data from early octopus.
+    # Please do not modify those JSON values.
+
+    spec = ServiceSpec.from_json(spec_json)
+    # just some verification that we can sill read old octopus specs
+    def convert_to_old_style_json(j):
+        j_c = dict(j.copy())
+        j_c.pop('service_name', None)
+        if 'spec' in j_c:
+            spec = j_c.pop('spec')
+            j_c.update(spec)
```

Comment:
This change seems wrong.

Reply:
There are two valid styles for service specs:

the "old" style:

```yaml
service_type: nfs
service_id: foo
pool: mypool
namespace: myns
```

and the "new" style:

```yaml
service_type: nfs
service_id: foo
spec:
  pool: mypool
  namespace: myns
```

In https://tracker.ceph.com/issues/45321 we decided that we'd like to prefer the new style as it is more readable and provides a better understanding of what fields are special for a give service type.

Unfortunately, we'll need to stay compatible with both versions for the the next two major releases.

```
spec = j_c.pop('spec')
j_c.update(spec)
```
converts the new style to the old style.

Maybe I should explain this somewhere in the code.